packages/core/src/radio-group/radio.ts

```diff
@@ -0,0 +1,72 @@
+import type { CSSResultGroup, PropertyValues, TemplateResult } from 'lit';
+import type { IElementInternals } from 'element-internals-polyfill';
+import { LitElement, html } from 'lit';
+import { property } from 'lit/decorators.js';
+import styles from './radio.scss.js';
+
+export class Radio extends LitElement {
+  /** @private */
+  static override styles: CSSResultGroup = styles;
+
+  #onClick = (event: Event): void => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    this.checked = true;
+  };
+
+  protected internals: ElementInternals & IElementInternals;
+
+  /** Whether the radio is selected. */
+  @property({ type: Boolean, reflect: true }) checked = false;
+
+  /** Wether this radio is disabled. */
+  @property({ type: Boolean, reflect: true }) disabled = false;
+
+  /** The value for this radio button. */
+  @property() value = '';
+
+  constructor() {
+    super();
+
+    this.internals = this.attachInternals();
+    this.internals.role = 'radio';
+  }
+
+  connectedCallback(): void {
+    super.connectedCallback();
+
+    this.addEventListener('click', this.#onClick);
+
+    if (!this.hasAttribute('tabindex')) {
+      this.tabIndex = 0;
+    }
+  }
+
+  disconnectedCallback(): void {
+    this.removeEventListener('click', this.#onClick);
+
+    super.disconnectedCallback();
+  }
+
+  updated(changes: PropertyValues<this>): void {
+    super.updated(changes);
+
+    if (changes.has('checked')) {
+      this.internals.ariaChecked = this.checked ? 'true' : 'false';
+
```

Comment:
Maybe checked radio should also have attribute `aria-checked = true`?

https://www.w3.org/WAI/ARIA/apg/patterns/radiobutton/
"If a radio button is checked, the radio element has [aria-checked](https://w3c.github.io/aria/#aria-checked) set to true. If it is not checked, it has [aria-checked](https://w3c.github.io/aria/#aria-checked) set to false."

Reply:
What do you mean? Using `ElementInternals`, it set's the ARIA attribute directly in the a11y tree. So there is no need to do `this.setAttribute('aria-checked', this.checked ? 'true' : 'false');`.